src/stream/src/from_proto/stream_scan.rs

```diff
@@ -147,11 +145,14 @@ impl ExecutorBuilder for StreamScanExecutorBuilder {
             }
             StreamScanType::Unspecified => unreachable!(),
         };
-        Ok(FlowControlExecutor::new(
-            executor,
+        let mut info = params.info.clone();
+        info.identity = format!("{} (flow controlled)", info.identity);
+
+        let exec = FlowControlExecutor::new(
+            (info, exec).into(),
             params.actor_context,
             node.rate_limit.map(|x| x as _),
-        )
-        .boxed())
+        );
+        Ok((params.info, exec).into())
```

Comment:
Seems if we don't wrap `FlowControlExecutor` direcly in `new_boxed_executor`, we can make it return `Box<dyn Execute>` instead of `Executor`, further reduce the possibility to return an inconsistent info.